File: Desafios_C_Sharp/Solução_de_Problemas_Essenciais_com_C_Sharp/desafio_05.cs
/* Desafio 5 - Pedro Bento e o Mundo de OZ */
using System;

class Desafio {
  
  static void Main(string[] args) {
    
    string input = "";
    string[] treasurer = new string[1];
    bool end = false;
    
    while(!end){
      
      input = Console.ReadLine();
      
      if(String.IsNullOrEmpty(input)){
        end = true;
      }
      else{
        if(!Array.Exists(treasurer, jewel => jewel == input)){
          treasurer[(treasurer.Length-1)] = input;
          Array.Resize(ref treasurer, (treasurer.Length+1));
        }
      }
    }
    Console.WriteLine(treasurer.Length-1);
  }
}

File: Desafios_Javascript/Solução_de_Problemas_Essenciais/desafio_05.js
/* Desafio 5 - Pedro Bento e o Mundo de OZ */

let input;
let treasurer = [];
let end = false;

while(!end){
  input = gets();
  
  if(input === ""){
    end = true;
  }
  else{
    if(!treasurer.includes(input)){
      treasurer.push(input);
    }
  }
}

console.log(treasurer.length)